src/main/java/uk/gov/hmcts/dts/fact/services/admin/AdminCourtAddressService.java

```diff
@@ -55,7 +65,57 @@ public AdminCourtAddressService(final CourtRepository courtRepository, final Cou
             .collect(toList());
     }
 
-    @SuppressWarnings("PMD.DataflowAnomalyAnalysis")
+    public List<String> validateCourtAddressPostcodes(final String slug, final List<CourtAddress> courtAddresses) {
+        final List<String> invalidPostcodes = new ArrayList<>();
+
+        if (!CollectionUtils.isEmpty(courtAddresses)) {
+            final List<String> allPostcodes = getAllPostcodesSortedByAddressType(courtAddresses);
+
+            if (!CollectionUtils.isEmpty(allPostcodes)) {
+                invalidPostcodes.addAll(validationService.validateFullPostcodes(allPostcodes));
+            }
+
+            // Only update the in-person court coordinates is all input postcodes are valid
+            if (invalidPostcodes.isEmpty()
+                && !CollectionUtils.isEmpty(allPostcodes)
+                && isInPersonCourt(slug)) {
+                updateCourtLatLonUsingPrimaryPostcode(slug, allPostcodes.get(0));
+            }
+        }
+        return invalidPostcodes;
+    }
+
+    private List<String> getAllPostcodesSortedByAddressType(final List<CourtAddress> courtAddresses) {
+        final Map<Integer, uk.gov.hmcts.dts.fact.entity.AddressType> addressTypeMap = addressTypeService.getAddressTypeMap();
+
+        // Sort the postcodes so that the 'visit us' or 'visit or contact us' postcodes appear first
+        return courtAddresses.stream()
+            .sorted(Comparator.comparingInt(a -> AddressType.isCourtAddress(getAddressTypeFromId(addressTypeMap, a.getAddressTypeId())) ? 0 : 1))
+            .map(CourtAddress::getPostcode)
+            .filter(StringUtils::isNotBlank)
+            .collect(toList());
+    }
+
+    private String getAddressTypeFromId(final Map<Integer, uk.gov.hmcts.dts.fact.entity.AddressType> map, final Integer addressTypeId) {
+        if (!map.containsKey(addressTypeId)) {
```

Comment:
It shouldn't happen for Admin Postal, but sometimes I like to treat the API to be independent from the admin portal and in that case, that should be checked in case sometimes passed in an incorrect ID